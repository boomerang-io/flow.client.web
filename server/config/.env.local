APP_ROOT=/BMRG_APP_ROOT_CONTEXT
APP_URL=localhost:3000
PORT=3000
NEW_RELIC_APP_NAME=
NEW_RELIC_LICENSE_KEY= 
HTML_HEAD_INJECTED_DATA=
HTML_HEAD_INJECTED_SCRIPTS=
CORE_ENV_URL=
BASE_APPS_ENV_URL=
CORE_SERVICE_ENV_URL=http://localhost:8000/api
PRODUCT_SERVICE_ENV_URL=http://localhost:8000/api

