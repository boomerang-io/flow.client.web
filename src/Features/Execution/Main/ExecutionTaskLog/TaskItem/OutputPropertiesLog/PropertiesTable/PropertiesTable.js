import React from "react";
import PropTypes from "prop-types";
import {
  StructuredListCell,
  StructuredListBody,
  StructuredListHead,
  StructuredListRow,
  StructuredListWrapper,
} from "@boomerang-io/carbon-addons-boomerang-react";
import { NoDisplay } from "@boomerang-io/carbon-addons-boomerang-react";
import styles from "./propertiesTable.module.scss";

function PropertiesTable({ data: properties, hasJsonValues = false }) {
  const formatPropertyValue = (value) => {
    if (hasJsonValues) {
      if (value && value !== '""')
        try {
          return JSON.parse(value);
        } catch {
          return "---";
        }
      return "---";
    } else {
      return value ?? "---";
    }
  };
  return (
    <div className={styles.tableContainer}>
      {properties && properties.length > 0 ? (
        <StructuredListWrapper selection>
          <StructuredListHead>
            <StructuredListRow head>
              <StructuredListCell head>Parameter</StructuredListCell>
              <StructuredListCell head>Value</StructuredListCell>
            </StructuredListRow>
          </StructuredListHead>
          <StructuredListBody>
            {properties.map((property, i) => (
              <StructuredListRow key={`row-${i}`}>
                <StructuredListCell>{property.key}</StructuredListCell>
                <StructuredListCell>
                  {<code className={styles.code}>{formatPropertyValue(property.value)}</code>}
                </StructuredListCell>
              </StructuredListRow>
            ))}
          </StructuredListBody>
        </StructuredListWrapper>
      ) : (
        <NoDisplay text="No parameters to display" />
      )}
    </div>
  );
}

PropertiesTable.propTypes = {
  data: PropTypes.array.isRequired,
};

export default PropertiesTable;
