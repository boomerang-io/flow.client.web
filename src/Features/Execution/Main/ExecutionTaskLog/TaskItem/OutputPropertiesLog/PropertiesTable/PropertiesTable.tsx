import React from "react";
import {
  StructuredListCell,
  StructuredListBody,
  StructuredListHead,
  StructuredListRow,
  StructuredListWrapper,
} from "@boomerang-io/carbon-addons-boomerang-react";
import { NoDisplay } from "@boomerang-io/carbon-addons-boomerang-react";
import styles from "./propertiesTable.module.scss";

type Props = {
  data: {
    key: string;
    value: string;
  }[] |
  {
    [key: string]: string;
  };
  hasJsonValues: boolean;
};

function PropertiesTable({ data: properties, hasJsonValues = false }: Props) {
  const formatPropertyValue = (value: string) => {
    if (hasJsonValues) {
      if (value && value !== '""')
        try {
          return JSON.parse(value);
        } catch {
          return "---";
        }
      return "---";
    } else {
      return value ?? "---";
    }
  };
  return (
    <div className={styles.tableContainer}>
      {properties && properties.length > 0 ? (
        <StructuredListWrapper selection>
          <StructuredListHead>
            <StructuredListRow head>
              <StructuredListCell head>Parameter</StructuredListCell>
              <StructuredListCell head>Value</StructuredListCell>
            </StructuredListRow>
          </StructuredListHead>
          <StructuredListBody>
            {Array.isArray(properties) && properties.map((property: {key: string; value: string;}, i: number) => (
              <StructuredListRow key={`row-${i}`}>
                <StructuredListCell>{property.key}</StructuredListCell>
                <StructuredListCell>
                  {<code className={styles.code}>{formatPropertyValue(property.value)}</code>}
                </StructuredListCell>
              </StructuredListRow>
            ))}
          </StructuredListBody>
        </StructuredListWrapper>
      ) : (
        <NoDisplay text="No parameters to display" />
      )}
    </div>
  );
}

export default PropertiesTable;
