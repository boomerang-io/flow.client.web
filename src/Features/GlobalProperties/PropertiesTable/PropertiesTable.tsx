import React, { useState } from "react";
import PropTypes from "prop-types";
import matchSorter from "match-sorter";
import { useMutation, useQueryClient } from "react-query";
import {
  DataTable,
  FeatureHeader as Header,
  FeatureHeaderTitle as HeaderTitle,
  FeatureHeaderSubtitle as HeaderSubtitle,
  notify,
  Pagination,
  Search,
  ToastNotification,
} from "@boomerang-io/carbon-addons-boomerang-react";
import ActionsMenu from "./ActionsMenu";
import CreateEditPropertiesModal from "./CreateEditPropertiesModal";
import { InputType, PASSWORD_CONSTANT } from "Constants";
import { formatErrorMessage } from "@boomerang-io/utils";
import EmptyState from "Components/EmptyState";
import { serviceUrl, resolver } from "Config/servicesConfig";
import { Property } from "Types";
import { Checkmark32, Close32 } from "@carbon/icons-react";
import styles from "./propertiesTable.module.scss";

const DEFAULT_PAGE_SIZE = 10;
const PAGE_SIZES = [DEFAULT_PAGE_SIZE, 25, 50];

const configUrl = serviceUrl.getGlobalConfiguration();

const defaultProperty = {
  value: "",
  readOnly: false,
  id: "",
  description: "",
  key: "",
  label: "",
  type: "",
};
function PropertiesTable({ properties }: { properties: Property[] }) {
  const queryClient = useQueryClient();

  const [page, setPage] = useState(1);
  const [pageSize, setPageSize] = useState(DEFAULT_PAGE_SIZE);
  const [searchQuery, setSearchQuery] = useState("");

  /** Delete Property */
  const { mutateAsync: deleteGlobalPropertyMutation } = useMutation(resolver.deleteGlobalPropertyRequest, {
    onSuccess: () => queryClient.invalidateQueries(configUrl),
  });

  const headers = [
    {
      header: "Label",
      key: "label",
      sortable: true,
    },
    {
      header: "Key",
      key: "key",
      sortable: true,
    },
    {
      header: "Description",
      key: "description",
      sortable: true,
    },
    {
      header: "Value",
      key: "value",
      sortable: true,
    },
    {
      header: "Secured",
      key: "secured",
      sortable: true,
    },
    {
      header: "",
      key: "actions",
    },
  ];

  const deleteProperty = async (property: Property) => {
    try {
      await deleteGlobalPropertyMutation({ id: property.id });
      notify(
        <ToastNotification
          kind="success"
          title={"Parameter Deleted"}
          subtitle={`Request to delete ${property.label} succeeded`}
          data-testid="delete-prop-notification"
        />
      );
    } catch (err) {
      const errorMessages = formatErrorMessage({
        error: err,
        defaultMessage: "Delete Parameter Failed",
      });
      notify(
        <ToastNotification
          kind="error"
          title={errorMessages.title}
          subtitle={errorMessages.message}
          data-testid="delete-prop-notification"
        />
      );
    }
  };

  const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const searchQuery = e.target.value;
    setSearchQuery(searchQuery);
  };

  const handlePaginationChange = ({ page, pageSize }: { page: number; pageSize: number }) => {
    setPage(page);
    setPageSize(pageSize);
  };

  const renderCell = (propertyId: string, cellIndex: number, value: any) => {
    const property = properties.find((property) => property.id === propertyId);
    const column = headers[cellIndex];

    switch (column.key) {
      case "value":
        const determineValue = value
          ? property && property.type === InputType.Password
            ? PASSWORD_CONSTANT
            : value
          : "---";
        return <p className={styles.tableTextarea}>{determineValue}</p>;
      case "secured":
        return property && property.type === InputType.Password ? (
          <Checkmark32 alt="secured" className={`${styles.tableSecured} ${styles.secured}`} />
        ) : (
          <Close32 alt="unsecured" className={`${styles.tableSecured} ${styles.unsecured}`} />
        );
      case "actions":
        return (
          <ActionsMenu deleteProperty={deleteProperty} property={property ?? defaultProperty} properties={properties} />
        );
      default:
        return <p className={styles.tableTextarea}>{value || "---"}</p>;
    }
  };

  const newProperties = searchQuery
    ? matchSorter(properties, searchQuery, { keys: ["label", "key", "description"] })
    : properties;

  const { TableContainer, Table, TableHead, TableRow, TableBody, TableCell, TableHeader } = DataTable;

  const totalItems = newProperties.length;

  return (
    <>
      <Header
        className={styles.header}
        includeBorder={false}
        header={
          <>
            <HeaderTitle className={styles.headerTitle}>Parameters</HeaderTitle>
            <HeaderSubtitle>Set global parameters that are available for all Workflows</HeaderSubtitle>
          </>
        }
      />
      <div className={styles.tableContainer}>
        <div className={styles.tableHeader}>
          <Search
            className={styles.search}
            id="parameters-table-search"
            labelText="Search"
            placeHolderText="Search"
            onChange={handleSearchChange}
          />
          <div className={styles.actions}>
            <CreateEditPropertiesModal properties={properties} />
          </div>
        </div>
        {totalItems > 0 ? (
          <>
            <DataTable
              rows={newProperties}
              headers={headers}
              render={({ rows, headers, getHeaderProps }: { rows: any; headers: any; getHeaderProps: any }) => (
                <TableContainer>
                  <Table isSortable>
                    <TableHead>
                      <TableRow className={styles.tableHeadRow}>
                        {headers.map((header: any) => (
                          <TableHeader
                            id={header.key}
                            {...getHeaderProps({
                              className: `${styles.tableHeadHeader} ${styles[header.key]}`,
                              header,
                              isSortable: header.sortable,
                            })}
                          >
                            {header.header}
                          </TableHeader>
                        ))}
                      </TableRow>
                    </TableHead>
                    <TableBody className={styles.tableBody}>
                      {rows.map((row: any) => (
                        <TableRow key={row.id} data-testid="configuration-parameter-table-row">
                          {row.cells.map((cell: any, cellIndex: number) => (
                            <TableCell key={cell.id}>
                              <div className={styles.tableCell}>{renderCell(row.id, cellIndex, cell.value)}</div>
                            </TableCell>
                          ))}
                        </TableRow>
                      ))}
                    </TableBody>
                  </Table>
                </TableContainer>
              )}
            />
            <Pagination
              onChange={handlePaginationChange}
              page={page}
              pageSize={pageSize}
              pageSizes={PAGE_SIZES}
              totalItems={totalItems}
            />
          </>
        ) : (
          <EmptyState title="Looks like there's nothing here" message="No results from that search, try again." />
        )}
      </div>
    </>
  );
}

PropertiesTable.propTypes = {
  properties: PropTypes.array,
};

export default PropertiesTable;
