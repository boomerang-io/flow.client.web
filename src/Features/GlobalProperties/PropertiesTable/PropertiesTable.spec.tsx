import React from "react";
import PropertiesTable from "./index";
// import config from "../../../apiServer/fixtures/config.js"
import config from "ApiServer/fixtures/config.js";

describe("PropertiesTable --- Snapshot Test", () => {
  it("Capturing Snapshot of PropertiesTable", () => {
    const { baseElement } = global.rtlRender(<PropertiesTable properties={config} />);
    expect(baseElement).toMatchSnapshot();
  });
});
