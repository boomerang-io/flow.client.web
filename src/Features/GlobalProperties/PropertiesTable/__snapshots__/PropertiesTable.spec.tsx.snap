// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertiesTable --- Snapshot Test Capturing Snapshot of PropertiesTable 1`] = `
<body
  id="app"
>
  <div>
    <header
      class="bx--bmrg-feature-header header"
    >
      <section
        class="bx--bmrg-feature-header__content"
      >
        <div
          class="bx--bmrg-feature-header__header"
        >
          <h1
            class="bx--bmrg-feature-header-title headerTitle"
          >
            Parameters
          </h1>
          <p
            class="bx--bmrg-feature-header-title"
          >
            Set global parameters that are available for all Workflows
          </p>
        </div>
      </section>
    </header>
    <div
      class="tableContainer"
    >
      <div
        class="tableHeader"
      >
        <div
          aria-labelledby="parameters-table-search-search"
          class="bx--search bx--search--xl search"
          role="search"
        >
          <div
            class="bx--search-magnifier"
          >
            <svg
              aria-hidden="true"
              class="bx--search-magnifier-icon"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15,14.3L10.7,10c1.9-2.3,1.6-5.8-0.7-7.7S4.2,0.7,2.3,3S0.7,8.8,3,10.7c2,1.7,5,1.7,7,0l4.3,4.3L15,14.3z M2,6.5	C2,4,4,2,6.5,2S11,4,11,6.5S9,11,6.5,11S2,9,2,6.5z"
              />
            </svg>
          </div>
          <label
            class="bx--label"
            for="parameters-table-search"
            id="parameters-table-search-search"
          >
            Search
          </label>
          <input
            autocomplete="off"
            class="bx--search-input"
            id="parameters-table-search"
            placeholder="Search"
            role="searchbox"
            type="text"
          />
          <button
            aria-label="Clear search input"
            class="bx--search-close bx--search-close--hidden"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
              />
            </svg>
          </button>
        </div>
        <div
          class="actions"
        >
          <button
            class="bx--btn bx--btn--md bx--btn--primary"
            data-testid="create-global-parameter-button"
            style="width: 12rem;"
            tabindex="0"
            type="button"
          >
            Create Parameter
            <svg
              aria-hidden="true"
              aria-label="Create Parameter"
              class="bx--btn__icon"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="bx--data-table-container"
      >
        <div
          class="bx--data-table-content"
        >
          <table
            class="bx--data-table bx--data-table--sort bx--data-table--no-border"
          >
            <thead>
              <tr
                class="tableHeadRow"
              >
                <th
                  aria-sort="none"
                  class="tableHeadHeader label"
                  id="label"
                  scope="col"
                >
                  <div
                    id="table-sort-2"
                    style="display: none;"
                  >
                    Click to sort rows by Label header in ascending order
                  </div>
                  <button
                    aria-describedby="table-sort-2"
                    class="tableHeadHeader label bx--table-sort"
                    type="button"
                  >
                    <span
                      class="bx--table-sort__flex"
                    >
                      <div
                        class="bx--table-header-label"
                      >
                        Label
                      </div>
                      <svg
                        aria-hidden="true"
                        class="bx--table-sort__icon"
                        fill="currentColor"
                        focusable="false"
                        height="20"
                        preserveAspectRatio="xMidYMid meet"
                        viewBox="0 0 32 32"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M16 4L6 14 7.41 15.41 15 7.83 15 28 17 28 17 7.83 24.59 15.41 26 14 16 4z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="bx--table-sort__icon-unsorted"
                        fill="currentColor"
                        focusable="false"
                        height="20"
                        preserveAspectRatio="xMidYMid meet"
                        viewBox="0 0 32 32"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                        />
                      </svg>
                    </span>
                  </button>
                </th>
                <th
                  aria-sort="none"
                  class="tableHeadHeader key"
                  id="key"
                  scope="col"
                >
                  <div
                    id="table-sort-3"
                    style="display: none;"
                  >
                    Click to sort rows by Key header in ascending order
                  </div>
                  <button
                    aria-describedby="table-sort-3"
                    class="tableHeadHeader key bx--table-sort"
                    type="button"
                  >
                    <span
                      class="bx--table-sort__flex"
                    >
                      <div
                        class="bx--table-header-label"
                      >
                        Key
                      </div>
                      <svg
                        aria-hidden="true"
                        class="bx--table-sort__icon"
                        fill="currentColor"
                        focusable="false"
                        height="20"
                        preserveAspectRatio="xMidYMid meet"
                        viewBox="0 0 32 32"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M16 4L6 14 7.41 15.41 15 7.83 15 28 17 28 17 7.83 24.59 15.41 26 14 16 4z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="bx--table-sort__icon-unsorted"
                        fill="currentColor"
                        focusable="false"
                        height="20"
                        preserveAspectRatio="xMidYMid meet"
                        viewBox="0 0 32 32"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                        />
                      </svg>
                    </span>
                  </button>
                </th>
                <th
                  aria-sort="none"
                  class="tableHeadHeader description"
                  id="description"
                  scope="col"
                >
                  <div
                    id="table-sort-4"
                    style="display: none;"
                  >
                    Click to sort rows by Description header in ascending order
                  </div>
                  <button
                    aria-describedby="table-sort-4"
                    class="tableHeadHeader description bx--table-sort"
                    type="button"
                  >
                    <span
                      class="bx--table-sort__flex"
                    >
                      <div
                        class="bx--table-header-label"
                      >
                        Description
                      </div>
                      <svg
                        aria-hidden="true"
                        class="bx--table-sort__icon"
                        fill="currentColor"
                        focusable="false"
                        height="20"
                        preserveAspectRatio="xMidYMid meet"
                        viewBox="0 0 32 32"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M16 4L6 14 7.41 15.41 15 7.83 15 28 17 28 17 7.83 24.59 15.41 26 14 16 4z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="bx--table-sort__icon-unsorted"
                        fill="currentColor"
                        focusable="false"
                        height="20"
                        preserveAspectRatio="xMidYMid meet"
                        viewBox="0 0 32 32"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                        />
                      </svg>
                    </span>
                  </button>
                </th>
                <th
                  aria-sort="none"
                  class="tableHeadHeader value"
                  id="value"
                  scope="col"
                >
                  <div
                    id="table-sort-5"
                    style="display: none;"
                  >
                    Click to sort rows by Value header in ascending order
                  </div>
                  <button
                    aria-describedby="table-sort-5"
                    class="tableHeadHeader value bx--table-sort"
                    type="button"
                  >
                    <span
                      class="bx--table-sort__flex"
                    >
                      <div
                        class="bx--table-header-label"
                      >
                        Value
                      </div>
                      <svg
                        aria-hidden="true"
                        class="bx--table-sort__icon"
                        fill="currentColor"
                        focusable="false"
                        height="20"
                        preserveAspectRatio="xMidYMid meet"
                        viewBox="0 0 32 32"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M16 4L6 14 7.41 15.41 15 7.83 15 28 17 28 17 7.83 24.59 15.41 26 14 16 4z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="bx--table-sort__icon-unsorted"
                        fill="currentColor"
                        focusable="false"
                        height="20"
                        preserveAspectRatio="xMidYMid meet"
                        viewBox="0 0 32 32"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                        />
                      </svg>
                    </span>
                  </button>
                </th>
                <th
                  aria-sort="none"
                  class="tableHeadHeader secured"
                  id="secured"
                  scope="col"
                >
                  <div
                    id="table-sort-6"
                    style="display: none;"
                  >
                    Click to sort rows by Secured header in ascending order
                  </div>
                  <button
                    aria-describedby="table-sort-6"
                    class="tableHeadHeader secured bx--table-sort"
                    type="button"
                  >
                    <span
                      class="bx--table-sort__flex"
                    >
                      <div
                        class="bx--table-header-label"
                      >
                        Secured
                      </div>
                      <svg
                        aria-hidden="true"
                        class="bx--table-sort__icon"
                        fill="currentColor"
                        focusable="false"
                        height="20"
                        preserveAspectRatio="xMidYMid meet"
                        viewBox="0 0 32 32"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M16 4L6 14 7.41 15.41 15 7.83 15 28 17 28 17 7.83 24.59 15.41 26 14 16 4z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="bx--table-sort__icon-unsorted"
                        fill="currentColor"
                        focusable="false"
                        height="20"
                        preserveAspectRatio="xMidYMid meet"
                        viewBox="0 0 32 32"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                        />
                      </svg>
                    </span>
                  </button>
                </th>
                <th
                  class="tableHeadHeader actions"
                  id="actions"
                  scope="col"
                />
              </tr>
            </thead>
            <tbody
              aria-live="polite"
              class="tableBody"
            >
              <tr
                data-testid="configuration-parameter-table-row"
              >
                <td>
                  <div
                    class="tableCell"
                  >
                    <p
                      class="tableTextarea"
                    >
                      Test parameter
                    </p>
                  </div>
                </td>
                <td>
                  <div
                    class="tableCell"
                  >
                    <p
                      class="tableTextarea"
                    >
                      test.key
                    </p>
                  </div>
                </td>
                <td>
                  <div
                    class="tableCell"
                  >
                    <p
                      class="tableTextarea"
                    >
                      adding a test description
                    </p>
                  </div>
                </td>
                <td>
                  <div
                    class="tableCell"
                  >
                    <p
                      class="tableTextarea"
                    >
                      Test value
                    </p>
                  </div>
                </td>
                <td>
                  <div
                    class="tableCell"
                  >
                    <svg
                      alt="unsecured"
                      aria-hidden="true"
                      class="tableSecured unsecured"
                      fill="currentColor"
                      focusable="false"
                      height="32"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 32 32"
                      width="32"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
                      />
                    </svg>
                  </div>
                </td>
                <td>
                  <div
                    class="tableCell"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="true"
                      aria-label="Overflow menu"
                      class="bx--overflow-menu bx--overflow-menu--md"
                      data-testid="configuration-parameter-table-overflow-menu"
                      type="button"
                    >
                      <svg
                        aria-label="Overflow menu icon"
                        class="bx--overflow-menu__icon"
                        fill="currentColor"
                        focusable="false"
                        height="16"
                        preserveAspectRatio="xMidYMid meet"
                        role="img"
                        viewBox="0 0 32 32"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <circle
                          cx="16"
                          cy="8"
                          r="2"
                        />
                        <circle
                          cx="16"
                          cy="16"
                          r="2"
                        />
                        <circle
                          cx="16"
                          cy="24"
                          r="2"
                        />
                        <title>
                          Overflow menu icon
                        </title>
                      </svg>
                    </button>
                  </div>
                </td>
              </tr>
              <tr
                data-testid="configuration-parameter-table-row"
              >
                <td>
                  <div
                    class="tableCell"
                  >
                    <p
                      class="tableTextarea"
                    >
                      ben.label
                    </p>
                  </div>
                </td>
                <td>
                  <div
                    class="tableCell"
                  >
                    <p
                      class="tableTextarea"
                    >
                      ben.key
                    </p>
                  </div>
                </td>
                <td>
                  <div
                    class="tableCell"
                  >
                    <p
                      class="tableTextarea"
                    >
                      ---
                    </p>
                  </div>
                </td>
                <td>
                  <div
                    class="tableCell"
                  >
                    <p
                      class="tableTextarea"
                    >
                      ---
                    </p>
                  </div>
                </td>
                <td>
                  <div
                    class="tableCell"
                  >
                    <svg
                      alt="secured"
                      aria-hidden="true"
                      class="tableSecured secured"
                      fill="currentColor"
                      focusable="false"
                      height="32"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 32 32"
                      width="32"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13 24L4 15 5.414 13.586 13 21.171 26.586 7.586 28 9 13 24z"
                      />
                    </svg>
                  </div>
                </td>
                <td>
                  <div
                    class="tableCell"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="true"
                      aria-label="Overflow menu"
                      class="bx--overflow-menu bx--overflow-menu--md"
                      data-testid="configuration-parameter-table-overflow-menu"
                      type="button"
                    >
                      <svg
                        aria-label="Overflow menu icon"
                        class="bx--overflow-menu__icon"
                        fill="currentColor"
                        focusable="false"
                        height="16"
                        preserveAspectRatio="xMidYMid meet"
                        role="img"
                        viewBox="0 0 32 32"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <circle
                          cx="16"
                          cy="8"
                          r="2"
                        />
                        <circle
                          cx="16"
                          cy="16"
                          r="2"
                        />
                        <circle
                          cx="16"
                          cy="24"
                          r="2"
                        />
                        <title>
                          Overflow menu icon
                        </title>
                      </svg>
                    </button>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="bx--pagination"
      >
        <div
          class="bx--pagination__left"
        >
          <label
            class="bx--pagination__text"
            for="bx-pagination-select-1"
            id="bx-pagination-select-1-count-label"
          >
            Items per page:
          </label>
          <div
            class="bx--form-item"
          >
            <div
              class="bx--select bx--select--inline bx--select__item-count"
            >
              <div
                class="bx--select-input--inline__wrapper"
              >
                <div
                  class="bx--select-input__wrapper"
                >
                  <select
                    class="bx--select-input"
                    id="bx-pagination-select-1"
                  >
                    <option
                      class="bx--select-option"
                      value="10"
                    >
                      10
                    </option>
                    <option
                      class="bx--select-option"
                      value="25"
                    >
                      25
                    </option>
                    <option
                      class="bx--select-option"
                      value="50"
                    >
                      50
                    </option>
                  </select>
                  <svg
                    aria-hidden="true"
                    class="bx--select__arrow"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <span
            class="bx--pagination__text bx--pagination__items-count"
          >
            1–2 of 2 items
          </span>
        </div>
        <div
          class="bx--pagination__right"
        >
          <div
            class="bx--form-item"
          >
            <div
              class="bx--select bx--select--inline bx--select__page-number"
            >
              <label
                class="bx--label bx--visually-hidden"
                for="bx-pagination-select-1-right"
              >
                Page number, of 1 pages
              </label>
              <div
                class="bx--select-input--inline__wrapper"
              >
                <div
                  class="bx--select-input__wrapper"
                >
                  <select
                    class="bx--select-input"
                    id="bx-pagination-select-1-right"
                  >
                    <option
                      class="bx--select-option"
                      value="1"
                    >
                      1
                    </option>
                  </select>
                  <svg
                    aria-hidden="true"
                    class="bx--select__arrow"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <span
            class="bx--pagination__text"
          >
            of 1 page
          </span>
          <div
            class="bx--pagination__control-buttons"
          >
            <button
              class="bx--pagination__button bx--pagination__button--backward bx--pagination__button--no-index bx--btn bx--btn--ghost bx--btn--disabled bx--tooltip--hidden bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--btn--icon-only--top bx--tooltip--align-center"
              disabled=""
              tabindex="0"
              type="button"
            >
              <div
                class="bx--assistive-text"
              >
                Previous page
              </div>
              <svg
                aria-hidden="true"
                aria-label="Previous page"
                class="bx--btn__icon"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                role="img"
                viewBox="0 0 32 32"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 24L10 16 20 8z"
                />
              </svg>
            </button>
            <button
              class="bx--pagination__button bx--pagination__button--forward bx--pagination__button--no-index bx--btn bx--btn--ghost bx--btn--disabled bx--tooltip--hidden bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--btn--icon-only--top bx--tooltip--align-end"
              disabled=""
              tabindex="0"
              type="button"
            >
              <div
                class="bx--assistive-text"
              >
                Next page
              </div>
              <svg
                aria-hidden="true"
                aria-label="Next page"
                class="bx--btn__icon"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                role="img"
                viewBox="0 0 32 32"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 8L22 16 12 24z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="bx--bmrg-modal-portal"
  />
  <div
    class="bx--bmrg-modal-portal"
  />
  <div
    class="bx--bmrg-modal-portal"
  />
  <div
    class="bx--bmrg-modal-portal"
  />
  <div
    class="bx--bmrg-modal-portal"
  />
</body>
`;
